packages/nns/src/governance.canister.ts

```diff
@@ -298,6 +299,74 @@ export class GovernanceCanister {
     return neuronId;
   };
 
+  // TODO: Rename to and replace `stakeNeuron` once `stakeNeuronIcrc1` is tested
+  // in NNS dapp.
+  /**
+   * @throws {@link InsufficientAmountError}
+   * @throws {@link StakeNeuronTransferError}
+   * @throws {@link CouldNotClaimNeuronError}
+   * @throws {@link TransferError}
+   */
+  public stakeNeuronIcrc1 = async ({
+    stake,
+    principal,
+    fromSubAccount,
+    icrcLedgerCanister,
+    createdAt,
+    fee,
+  }: {
+    stake: bigint;
+    principal: Principal;
+    fromSubAccount?: number[];
+    icrcLedgerCanister: IcrcLedgerCanister;
+    // Used for the TransferRequest parameters.
+    // Check the TransferRequest type for more information.
+    createdAt?: bigint;
+    fee?: E8s;
+  }): Promise<NeuronId> => {
+    if (stake < E8S_PER_TOKEN) {
+      throw new InsufficientAmountError(stake);
+    }
+
+    const nonceBytes = new Uint8Array(randomBytes(8));
+    const nonce = uint8ArrayToBigInt(nonceBytes);
+    const toSubAccount = this.getNeuronStakeSubAccountBytes(
```

Comment:
This is what we mentioned that might still create problems, right?

So, this changes won't fully eliminate the problem.

Reply:
Correct. We also need a new method before we can fix that. There's a [ticket](https://dfinity.atlassian.net/browse/NNS1-2606) for that for the NNS team.